src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -0,0 +1,138 @@
+(function() {
+  goog.provide('ga_backgroundselector_directive');
+
+  goog.require('ga_map');
+  goog.require('ga_permalink');
+
+  var module = angular.module('ga_backgroundselector_directive', [
+    'ga_map',
+    'ga_permalink',
+    'pascalprecht.translate'
+  ]);
+
+  module.directive('gaBackgroundSelector',
+    function($document, gaPermalink, gaLayers, gaLayerFilters,
+      gaBrowserSniffer) {
+      return {
+        restrict: 'A',
+        templateUrl:
+            'components/backgroundselector/partials/backgroundselector.html',
+        scope: {
+          map: '=gaBackgroundSelectorMap'
+        },
+        link: function(scope, elt, attrs) {
+          scope.isBackgroundSelectorClosed = true;
+          scope.embed = gaBrowserSniffer.embed;
+          var map = scope.map;
+          var firstLayerChangeEvent = true;
```

Comment:
Slang, sorry. it's guarding against multiple runs of some code.

Anyway, it was the same in old code. I just said that you don't need it because currentTopic is basically doing exactly the same thing (currentTopic will be undefined when 'gaLayersChange' event is first triggered, after that it will be set. So below, instead of using `if (firstLayerChangeEvent)`, you could use `if (!scope.curentTopic)` thus saving one variable. It's minor.